tests/optional_kernel_features/kernel_features_mismatched_nd_range_exception.cpp

```diff
@@ -0,0 +1,100 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Provides tests for the exception that is thrown by
+//  [[sycl::reqd_work_group_size]] attribute when use nd_range of wrong size in
+//  kernel.
+//
+*******************************************************************************/
+
+#include "../common/common.h"
+#include "catch2/catch_template_test_macros.hpp"
+
+namespace kernel_features_mismatched_nd_range_exception {
+using namespace sycl_cts;
+
+// Define required size of work group for attribute
+constexpr int testing_wg_size = 16;
+
+class Functor {
+ public:
+  [[sycl::reqd_work_group_size(testing_wg_size)]] void operator()(
+      sycl::item<1>) const {}
+};
+
+TEST_CASE(
+    "Kernel features. Exceptions throwns by [[reqd_work_group_size()]] with "
+    "mismatched nd_range",
+    "[kernel_features]") {
+  auto queue = util::get_cts_object::queue();
+  const auto max_wg_size =
+      queue.get_device().get_info<sycl::info::device::max_work_group_size>();
+  if (max_wg_size < testing_wg_size) {
+    WARN("Device supported work group size too small. Skipping..");
+    return;
+  }
+
+  const bool is_exception_expected = true;
+  sycl::errc errc_expected = sycl::errc::nd_range;
```

Comment:
From the spec:
> In order to guarantee source code portability of SYCL applications that use optional kernel features, all SYCL implementations must be able to compile device code that uses these optional features regardless of whether the implementation supports the features on any of its devices.

You've marked two implementations in the CI as not passing (adding the test to the filter), but I believe they should be able to pass the test even if they don't implement the features, otherwise they wouldn't be able to pass conformance in the current CTS design.

Reply:
As I understand CI doesn't run tests, it's just built it. [Here](https://github.com/KhronosGroup/SYCL-CTS/runs/5209986361?check_suite_focus=true) you can find output from the building of tests on CI. ComputeCpp and hipsycl failed building due to some errors.

From hipsycl logs, line 1970: `error: ‘errc’ is not a member of ‘sycl’`. Looks like enum with error codes is not implemented yet and attributes too (line 1966).

Similar situation for computecpp.

But maybe I was wrong when I put these two implementations in the filter because both of them give the error message: `no known conversion from 'cl::sycl::nd_item<1>' to 'sycl::item<1>'` error, so it can be bug of tests itself.

Reply:
It was my mistake. I changed to `nd_item` but now getting another error. Right now investigating it.